### PR TITLE
Added AMD support

### PR DESCRIPTION
**Pull request by uzikilon**
- Cleaned up the factory loader
- Tweaked a bunch of tests where the date was not parsing correctly
- Tuned the the .jshintrc file

The date failing test might actually be a problem in the date parser. I haven't looked into it, but changed the time to one that doesn't fail. It might be worth looking into.

### CODE REVIEW

**Comment by uzikilon** [comment]
@dnemoga Do you approve this commit?

**Comment by alexpusch** [comment]
Thanks for the enhancement!

Don't edit the dist files. If you do, other PR's will be unmergeable. Update the src files. Rebase your commits so there will be only one commit with changes to src.

File: src/backbone/schema.js [hunk 1]
```diff
@@ -465,7 +458,7 @@
                         ////////////////////
 
                         if (_.isUndefined(value)) {
-                            value = self.defaultValue(attribute);
+                            value = this.defaultValue(attribute);
```
Comment (alexpusch) [inline src/backbone/schema.js]: Look at https://github.com/dnemoga/Backbone.Schema/pull/17/files, I've done this there.

**Comment by uzikilon** [comment]
@alexpusch updated

**Comment by alexpusch** [comment]
Please remove the dist/*  files changes from your commit. I would update them on next version release.

**Comment by uzikilon** [comment]
Sorry. fixed.

**Pull request merged**